soc/nxp/imxrt/imxrt11xx/soc.c

```diff
@@ -730,7 +730,15 @@ static int imxrt_init(void)
 }
 
 #ifdef CONFIG_SOC_RESET_HOOK
-void soc_reset_hook(void)
+__asm__ (
```

Comment:
I think we should add a comment here explaining why this code needs to exist, something like:
```
Stack pointer is not set at this point in the early init, but we call C functions from the SOC reset.
Set a stack pointer so that C functions will work correctly
```